ouroboros-network/src/Ouroboros/Network/Snocket.hs

```diff
@@ -129,3 +134,115 @@ data Snocket m fd addr = Snocket {
 
   , toBearer      :: Tracer m MuxTrace -> fd -> (MuxBearer m)
   }
+
+
+socketAddrFamily
+    :: Socket.SockAddr
+    -> AddressFamily Socket.SockAddr
+socketAddrFamily (Socket.SockAddrInet  _ _    ) = SocketFamily Socket.AF_INET
+socketAddrFamily (Socket.SockAddrInet6 _ _ _ _) = SocketFamily Socket.AF_INET6
+socketAddrFamily (Socket.SockAddrUnix _       ) = SocketFamily Socket.AF_UNIX
+
+type SocketSnocket = Snocket IO Socket SockAddr
+
+-- | Create a 'Snocket' for the given 'Socket.Family'. In the 'bind' method set
+-- 'Socket.ReuseAddr` and 'Socket.ReusePort'.
+--
+socketSnocket
+  :: AssociateWithIOCP
+  -- ^ associate the socket with I/O CP.  We use it when we connect (it can be
+  -- used multiple times on a socket) and when we accept a socket.
+  --
+  -- Though it could be used in `open`, but that is going to be used in
+  -- a bracket so it's better to keep it simple.
+  --
+  -> SocketSnocket
+socketSnocket iocp = Snocket {
+      getLocalAddr   = Socket.getSocketName
+    , getRemoteAddr  = Socket.getPeerName
+    , addrFamily     = socketAddrFamily
+    , open           = openSocket
+    , openToConnect  = \addr -> openSocket (socketAddrFamily addr)
+    , connect        = \s a -> do
+#if !defined(mingw32_HOST_OS)
+        Socket.connect s a
+#else
+        Win32.Async.connect s a
+#endif
+    , bind = \sd addr -> do
+        let SocketFamily fml = socketAddrFamily addr
+        when (fml == Socket.AF_INET ||
+              fml == Socket.AF_INET6) $ do
+          Socket.setSocketOption sd Socket.ReuseAddr 1
+#if !defined(mingw32_HOST_OS)
+          -- not supported on Windows 10
+          Socket.setSocketOption sd Socket.ReusePort 1
+#endif
+        when (fml == Socket.AF_INET6)
+          -- An AF_INET6 socket can be used to talk to both IPv4 and IPv6 end points, and
+          -- it is enabled by default on some systems. Disabled here since we run a separate
+          -- IPv4 server instance if configured to use IPv4.
+          $ Socket.setSocketOption sd Socket.IPv6Only 1
+
+        Socket.bind sd addr
+    , listen   = \s -> Socket.listen s 1
```

Comment:
A leftover from the old code, but we should increase the listening queue. Also in `rawSocketSnocket`.

Reply:
I actually wanted to ask you, what the value should be? Great that you picked it yourself :)

Reply:
8 sounds good to me, we can revisit it once we have an overall limit on the number of open file descriptors.

Reply:
Done